### ISSUE TITLE
Wrong label annotations when training on custom dataset

### ISSUE THREAD

**Issue by Nozomeister** [WongKinYiu/YOLO#102]
Hello, 
First of all, thank you for the MIT license implementation of YOLOv9.
I want to report an issue when training with a custom dataset.
I am using the WIDER dataset and I have converted the annotations to YOLO format i.e. xywh.
However, when I attempted to train the yolov9-c model on the dataset, I found out that the annotations are wrong from the visualized ground truth, thus the model was trained wrongly with incorrect label loaded. I have attached a screenshot of the labels on ground truth visualized by wandb below.
My training script is:
`task.epoch=100 dataset=widerface task.data.batch_size=8 model=v9-c device=cuda name=face use_wandb=True`

The visualization of the annotated GT is attached as below:
![image](https://github.com/user-attachments/assets/c09bbc5a-704e-4be6-8dca-608868a70ce4)

I investigated the situation and figure out there might be an issue when loading the bbox in the file [yolo/tools/data_loader.py](url) line 125:
`bbox = torch.tensor([cls, *valid_points.min(axis=0), *valid_points.max(axis=0)])
`

When debugging, bbox is returned as a list under [c w h x y] and not [c x y w h], where c is the class number. I attempted to reverse it to [c x y w h], but when loading the new dataset cache and training, the loss approaching NaN and infinity.

Could you help me with this issue? Thank you and best regards.

**Comment by nkinnaird** [comment]
I'm not sure if this is fixed on another branch or I have different config settings somewhere, but in the `data_loader.py` file there are these lines:

```
    def __getitem__(self, idx) -> Tuple[Image.Image, Tensor, Tensor, List[str]]:
        img, bboxes, img_path = self.get_data(idx)
        img, bboxes, rev_tensor = self.transform(img, bboxes)
        bboxes[:, [1, 3]] *= self.image_size[0] # Nick - looks like boxes are scaled up to pixel values
        bboxes[:, [2, 4]] *= self.image_size[1]
        return img, bboxes, rev_tensor, img_path
```

which appear to scale the bounding box coordinates up to the pixel coordinates, and as defined assumes that the points are in `x_min, y_min, x_max, y_max` format, rather than either yolo format (`x_center`, `y_center`, `width`, `height`) or coco format (`x_min`, `y_min`, `width`, `height`). I'm using yolo txt files so perhaps there is some other code that converts coco json files to the right format, not sure.

Regardless making this change in the `load_valid_labels` method to convert my yolo labels fixes the issues and the ground truth annotations look correct when viewed in WandB:

```
        bboxes = []
        for seg_data in seg_data_one_img:
            cls = seg_data[0]
            # points = np.array(seg_data[1:]).reshape(-1, 2) # original

            x, y, w, h = seg_data[1:]
            x_min = x - w/2
            x_max = x + w/2
            y_min = y - h/2
            y_max = y + h/2
            points = np.array([x_min, y_min, x_max, y_max]).reshape(-1, 2)
```

Maybe that will work for you or you can make a similar change based on your data.

**Comment by sergiosanchoasensio** [comment]
@Nozomeister were you able to resolve the issue? I'm encountering the same problem. My dataset originally used the bounding box format (x1, y1, x2, y2). I converted the annotations to YOLO format (x_center, y_center, width, height), but when visualizing the ground truth in WandB, the bounding boxes are not placed correctly. I also attempted converting the annotations to COCO format (x1, y1, width, height), but the issue persists.

**Comment by luuzk** [comment]
This is still an open issue since this implementation of YOLOv9 cannot read object detection labels in YOLO format
